affinidi-messaging-mediator/tests/integration_test.rs

```diff
@@ -0,0 +1,860 @@
+use affinidi_did_resolver_cache_sdk::{config::ClientConfigBuilder, DIDCacheClient};
+use affinidi_messaging_didcomm::{
+    secrets::{Secret, SecretsResolver},
+    Message, PackEncryptedOptions,
+};
+use affinidi_messaging_mediator::{resolvers::affinidi_secrets::AffinidiSecrets, server::start};
+use affinidi_messaging_sdk::{
+    config::Config,
+    conversions::secret_from_str,
+    errors::ATMError,
+    messages::{
+        fetch::FetchOptions, sending::InboundMessageResponse, AuthenticationChallenge,
+        AuthorizationResponse, DeleteMessageRequest, DeleteMessageResponse, Folder,
+        GenericDataStruct, GetMessagesRequest, GetMessagesResponse, MessageList,
+        MessageListElement, SuccessResponse,
+    },
+    transports::SendMessageResponse,
+};
+use common::{
+    BOB_DID, BOB_E1, BOB_V1, CONFIG_PATH, MEDIATOR_API, MY_DID, MY_E1, MY_V1, SECRETS_PATH,
+};
+use core::panic;
+use message_builders::{
+    build_delivery_request_message, build_forward_request_message, build_message_received_message,
+    build_ping_message, build_status_request_message, create_auth_challenge_response,
+};
+use reqwest::{Certificate, Client, ClientBuilder};
+use response_validations::{
+    validate_forward_request_response, validate_get_message_response, validate_list_messages,
+    validate_message_delivery, validate_message_received_status_reply, validate_status_reply,
+};
+use sha256::digest;
+use std::{
+    fs::{self, File},
+    io::{self, BufRead, BufReader},
+    path::Path,
+    process::Command,
+    str,
+    time::Duration,
+};
+use tokio::time::sleep;
+
+mod common;
+mod message_builders;
+mod response_validations;
+
+#[tokio::test]
+async fn test_mediator_server() {
+    // Generate secrets and did for mediator if not existing
+    if !fs::metadata(SECRETS_PATH).is_ok() {
+        println!("Generating secrets");
+        _generate_keys();
+        _generate_secrets();
+        let mediator_did = _get_did_from_secrets(SECRETS_PATH.into());
+        _inject_did_into_config(CONFIG_PATH, &mediator_did);
+        println!("Secrets generated and did injected to mediator.toml");
+    }
+
+    _start_mediator_server().await;
+
+    // Allow some time for the server to start
+    sleep(Duration::from_millis(2000)).await;
+
+    let config = Config::builder()
+        .with_ssl_certificates(&mut vec![
+            "../affinidi-messaging-mediator/conf/keys/client.chain".into(),
+        ])
+        .build()
+        .unwrap();
+
+    let did_resolver = DIDCacheClient::new(ClientConfigBuilder::default().build())
+        .await
+        .unwrap();
+    let my_secrets_resolver = AffinidiSecrets::new(vec![
```

Comment:
WDYT to use alice_secrets_resolver, as it is used as usual to illustrate communication between 2 actors (Alice and Bob)